### ISSUE TITLE
Create mixins for Angular table column directives 

### ISSUE THREAD

**Issue by m-akinc** [ni/nimble#1211]
## 🧹 Tech Debt
Multiple column types are likely to mix-in the FractionalWidthColumnAPI and/or the GroupableColumnAPI. The Angular directives for those column types will also expose those APIs. We should have mixins for those directives. Once we have them, we should update the Text and Anchor column types to use them.

**Comment by rajsite** [comment]
Re-opening as #1587 was reverted in #1591
An [Angular issue](https://github.com/angular/angular/issues/42594#issuecomment-868091258) describes why using TypeScript Mixins does not work for Angular Directives and recommends checking out how Angular Material handles mixins. I didn't look but we can check what Angular Material does and see if it's useful for us.

**Comment by mollykreis** [comment]
tldr - After looking into the approach that Angular Material uses, it doesn't see particularly useful for us. Closing this as not planned because there isn't a clear path forward to is significantly better than our current approach of copy/pasting between directives.

The approach taken in Angular Material is to mixin a base class that isn't a `@Directive` and doesn't contain any properties marked as `@Input`. Instead, that mixed in base class provides only the TS implementation of the getter(s) and setter(s) for properties. Then, the actual `@Directive` has to configure an `inputs` array as part of its decorator, which in our case also has to rename each input to follow our naming conventions. Therefore, our implementation would need to look something like the following:

```ts
@Directive({
    selector: 'nimble-table-column-text',
    // eslint-disable-next-line @angular-eslint/no-inputs-metadata-property, @angular-eslint/no-input-rename
    inputs: ['fractionalWidth: fractional-width', 'minPixelWidth: min-pixel-width', 'groupIndex: group-index', 'groupingDisabled: grouping-disabled']
})
export class NimbleTableColumnTextDirective extends mixinFractionalWidthColumnAPI(mixinGroupableColumnAPI(NimbleTableColumnBaseDirective<TableColumnText>)) {
    /* implementation specific to the text column */
}
```

It's not ideal to have to call the mixin function and still have to know all the properties & their mapped inputs. This also doesn't allow a mixin to add a new input without having to update each usage of that mixin. I attempted to single-source the input string within the mixin file, but that didn't work because the mixin is in a different entry point (`table-column` rather than `table-column/text`); therefore, the string array isn't statically known at compile time like is required.

Additionally, the above approach didn't actually work because all of the `@Input`s coming from `NimbleTableColumnBaseDirective` were lost from the type. This may be a solvable problem, but we decided that it wasn't worth further attempts to solve it since the general direction of the approach already seemed to be adding significant complexity to our code while still requiring a decent amount of code duplication across each directive.
